includes/class-wc-payments-fraud-service.php

```diff
@@ -195,4 +196,46 @@ private function get_cookie_session_id() {
 		$cookie_customer_id = $cookie[0];
 		return $wpcom_blog_id . '_' . $cookie_customer_id;
 	}
+
+	/**
+	 * Adds the Sift JS page tracker if needed. See the comments for the detailed logic.
+	 *
+	 * @return  void
+	 */
+	public function add_sift_js_tracker() {
+		if ( ! isset( $_GET['wcpay-connection-success'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			// Only enqueue the tracker if the merchant has just finished the Stripe KYC.
+			return;
+		}
+
+		$fraud_config = $this->account->get_fraud_services_config();
+		if ( ! isset( $fraud_config['sift'] ) ) {
+			// Abort if Sift is not enabled for this account.
+			return;
+		}
+
+		if ( isset( $_GET['path'] ) && strpos( $_GET['path'], '/payments/' ) === 0 ) { // phpcs:ignore WordPress.Security
+			// If the current page is a WCPay dashboard, there's separate logic that will include the Sift JS tracker on its own.
+			return;
+		}
+
+		?>
+		<script type="text/javascript">
+			var src = 'https://cdn.sift.com/s.js';
+
+			var _sift = ( window._sift = window._sift || [] );
+			_sift.push( [ '_setAccount', '<?php echo esc_attr( $fraud_config['sift']['beacon_key'] ); ?>' ] );
+			_sift.push( [ '_setUserId', '<?php echo esc_attr( $fraud_config['sift']['user_id'] ); ?>' ] );
+			_sift.push( [ '_setSessionId', '<?php echo esc_attr( $fraud_config['sift']['session_id'] ); ?>' ] );
+			_sift.push( [ '_trackPageview' ] );
+
+			if ( ! document.querySelector( '[src="' + src + '"]' ) ) {
+				var script = document.createElement( 'script' );
+				script.src = src;
+				script.async = true;
+				document.body.appendChild( script );
+			}
+		</script>
```

Comment:
Yeah, I saw that duplication, but to be honest I think adding a separate entry point just for this is overkill. I agree that's the only solution to de-duplicate this code because we still need to enqueue the fraud scripts from the React side.

It's 12 lines of code, and and it won't get larger over time (we've decided on Sift, and Stripe doesn't need this extra tracking data). Are you ok merging this as-is?

Reply:
> Or we could maybe use this PHP function across all of the pages instead of the React one.

I thought about that but it's not possible, because of client-side navigation. If a user is on `WooCommerce -> Home` and goes to `Payments -> Overview`, that page view needs to be tracked but there was no PHP page load at all.

> Fair enough. We can revisit once the logic if it ever grows.

Thanks!